packages/roosterjs-editor-plugins/lib/plugins/Picker/PickerPlugin.ts

```diff
@@ -53,18 +53,65 @@ const UNIDENTIFIED_CODE = [0, 229];
  */
 export default class PickerPlugin<T extends PickerDataProvider = PickerDataProvider>
     implements EditorPlugin {
-    private editor: IEditor;
-    private eventHandledOnKeyDown: boolean;
-    private blockSuggestions: boolean;
-    private isSuggesting: boolean;
-    private lastKnownRange: Range;
+    private editor: IEditor | null = null;
+    private core: PickerPluginCore | null = null;
+    public dataProvider: T | undefined = undefined;
+    private pickerOptions: PickerPluginOptions | undefined = undefined;
+
+    constructor(dataProvider: T | undefined, pickerOptions: PickerPluginOptions) {
+        this.dataProvider = dataProvider;
+        this.pickerOptions = pickerOptions;
+    }
+
+    public willHandleEventExclusively(event: PluginEvent) {
+        return this.core ? this.core.willHandleEventExclusively(event) : false;
+    }
+
+    public initialize(editor: IEditor) {
+        this.editor = editor;
+        if (this.editor && this.dataProvider && this.pickerOptions) {
+            this.core = new PickerPluginCore(this.editor, this.dataProvider, this.pickerOptions);
+            this.core.initialize();
+        }
+    }
+
+    public onPluginEvent(event: PluginEvent) {
+        if (this.core) {
+            this.core.onPluginEvent(event);
+        }
+    }
+
+    public dispose() {
+        this.editor = null;
+        this.dataProvider = undefined;
+        this.pickerOptions = undefined;
+        if (this.core) {
+            this.core.dispose();
+            this.core = null;
+        }
+    }
+
+    public getName() {
+        return 'Picker';
+    }
+}
+
+class PickerPluginCore<T extends PickerDataProvider = PickerDataProvider> implements EditorPlugin {
```

Comment:
Actually, this class doesn't need to implement EditorPlugin interface, and that is why I suggest merge constructor and initialize function together.

Reply:
I see, I got it now.